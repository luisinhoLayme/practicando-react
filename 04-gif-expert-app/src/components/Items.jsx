import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {
  // console.log(title, url)
  return (
    <div>
      <strong>{title}</strong>
      <img src={url} alt={title} />
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string
}
