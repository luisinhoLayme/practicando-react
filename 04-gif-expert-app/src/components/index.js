
export * from './Loading'
export * from './Items'
export * from './AddCategory'
export * from './GifGrid'

