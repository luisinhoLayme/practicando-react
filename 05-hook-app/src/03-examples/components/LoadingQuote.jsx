
export const LoadingQuote = () => {
  return (
    <div>
      Loading...
    </div>
  )
}
