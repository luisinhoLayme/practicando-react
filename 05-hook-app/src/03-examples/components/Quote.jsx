import PropTypes from 'prop-types'
import { useState } from 'react'
import { useRef } from 'react'
import { useLayoutEffect } from 'react'

export const Quote = ({ quote, author }) => {

  const pRef = useRef()
  const [boxSize, setBoxSize] = useState({ width: 0, height: 0 })

  useLayoutEffect(() => {
    const {width, height} = pRef.current.getBoundingClientRect()
    setBoxSize({width, height})

  }, [quote])

  return (
    <>
      <blockquote>
        <p>{quote}</p>
        <footer style={{ color: '#aaa', textAlign: 'right' }}>- <span ref={pRef}>{author}</span></footer>
      </blockquote>
      <code>{JSON.stringify(boxSize)}</code>
    </>
  )
}

Quote.propTypes = {
  quote: PropTypes.string,
  author: PropTypes.string
}
