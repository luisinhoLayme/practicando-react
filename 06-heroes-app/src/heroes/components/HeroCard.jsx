import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import '../styles/hero-card.css'

export const HeroCard = ({
  id,
  superhero,
  alter_ego,
  first_appearance,
  characters
}) => {

  const path = `/heroes/${ id }.jpg`

  return (
    <article
      key={id}
      className="card animate__animated animate__fadeIn"
    >
      <img src={path} alt={superhero} className="card__img" />
      <aside className="card__info">
        <header><h3>{ superhero }</h3></header>
        <p className="card__paragraph">{ alter_ego }</p>

        {characters !== alter_ego && <p className="card__paragraph">{ characters }</p>}

        <footer className="card__footer">{ first_appearance }</footer>
        <Link className="card__link" to={`/hero/${ id }`}>Mas...</Link>
      </aside>

    </article>
  )
}

HeroCard.propTypes = {
  id: PropTypes.string,
  superhero: PropTypes.string,
  publisher: PropTypes.string,
  alter_ego: PropTypes.string,
  first_appearance: PropTypes.string,
  characters: PropTypes.string,
}
