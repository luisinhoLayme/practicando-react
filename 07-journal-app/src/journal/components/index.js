
export * from './Navbar'
export * from './Sidebar'
export * from './ImageGalery'
export * from './SideBarItem'
