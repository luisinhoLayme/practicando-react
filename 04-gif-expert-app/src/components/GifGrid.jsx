import PropTypes from 'prop-types'
import { GifItem } from './Items'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { Loading } from './Loading'

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category )

  return (
    <div>
      <h3>{category}</h3>
      { isLoading && <Loading />}
      {/* {isLoading && <div>Loading...</div>} */}

      <div>
        {images.map(({id, ...rest}) => (
          <GifItem
            key={ id }
            {...rest}
          />
        ))}
      </div>
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
