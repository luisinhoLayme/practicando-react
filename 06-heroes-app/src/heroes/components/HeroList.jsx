import { useMemo } from 'react'
import { getHeroesByPublisher } from '../helpers'
import PropTypes from 'prop-types'
import { HeroCard } from './HeroCard'

export const HeroList = ({ publisher }) => {

  const heroes = useMemo(() => getHeroesByPublisher( publisher ), [publisher])

  return (
    <section className="container">
      {heroes.map((hero) => (
        <HeroCard
          key={hero.id}
          {...hero}
        />
      ))}
    </section>
  )
}

HeroList.propTypes = {
  publisher: PropTypes.string
}


