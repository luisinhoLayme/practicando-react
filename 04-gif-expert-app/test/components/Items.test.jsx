import { render, screen } from "@testing-library/react"
import { GifItem } from '../../src/components/Items'

describe('Pruebas en <Items.jsx />', () => {

  const title = 'Luisinho'
  const url = 'https://luisinho.dev/luisinho.jpg'

  test('debe de hacer el snapshot del componente', () => {

    const { container } = render(
      <GifItem
        title={title}
        url={url}
      />
    )
    expect(container).toMatchSnapshot()

  });

  test('debe de mostrar la imagen con el URL y al ALT indicado', () => {

    render(<GifItem title={title} url={url} />)
    // screen.debug()
    // expect( screen.getByRole('img').src ).toBe( url )
    // expect( screen.getByRole('img').alt ).toBe( title )
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( alt )

  })

  test('debe de mostrar el titulo en el componente', () => {

    render(<GifItem title={title} url={url} />)
    expect( screen.getByText( title ) ).toBeTruthy()

  })
})

