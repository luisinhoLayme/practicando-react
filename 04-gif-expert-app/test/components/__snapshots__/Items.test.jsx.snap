// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Items.jsx /> debe de hacer el snapshot del componente 1`] = `
<div>
  <div>
    <strong>
      Luisinho
    </strong>
    <img
      alt="Luisinho"
      src="https://luisinho.dev/luisinho.jpg"
    />
  </div>
</div>
`;
